superset/models/core.py

```diff
@@ -761,8 +761,8 @@ def wrapper(*args, **kwargs):
             post_data = request.form or {}
             d.update(post_data)
             d.update(kwargs)
-            slice_id = d.get('slice_id', 0)
             try:
+                slice_id = d.get('slice_id') or json.loads(d.get('form_data')).get('slice_id')
```

Comment:
This will not catch when you don't have nor slice_id neither form_data:
https://travis-ci.org/apache/incubator-superset/jobs/266166713#L4907